researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -561,6 +562,14 @@ public void addLocalDataGroup(@NonNull String dataGroup) {
     public List<String> getLocalDataGroups() {
         logger.debug("Called getLocalDataGroups");
 
+        // The user session info data groups field is the most up to date reflection of data groups
+        UserSessionInfo userSessionInfo = bridgeManagerProvider.getAuthenticationManager().getUserSessionInfo();
```

Comment:
I figured this was a reasonable assumption to make, lmk if I'm wrong, however.  The AccountDAO was unpredictable for getting data groups.

Reply:
Could you clarify? `accountManager.getUserSessionInfo()` delegates to `accountDAO.getUserSessionInfo()` so it seems the predictability should be the same.

Reply:
You would think that, but accountDao.getDataGroups actually references a completely different shared pref field...

`List<String> rawDataGroupList = getValue(KEY_DATA_GROUPS, STRING_LIST);`

Reply:
Should I switch accountDao.getDataGroups over to just read the user session?

Reply:
OH, missed that this is getLocalDataGroups. That was developed for CRF to let a DataGroupQuestionStep temporarily store a local group. It's used in conjunction with a `shouldPersist` flag to decide when/where to sync to Bridge.

accountDao.getDataGroups perhaps should be moved out of DAO which is geared towards long-term persistence. I think storing in memory in BridgeDataProvider would be a lot less confusing

Reply:
Oh! Hmmm.  Maybe for this particular problem I'll get the user session from BridgeDataProvider instead and read the data groups.